app/services/importers/okc_blotter/pdf.rb

```diff
@@ -0,0 +1,139 @@
+require 'zip'
+
+module Importers
+  module OkcBlotter
+    class Pdf < ApplicationService
+      def initialize(date)
+        @date = date
+      end
+
+      def perform
+        download_pdf_from_website unless pdf
+        save
+      end
+
+      def download_pdf_from_website
+        input = HTTParty.get("#{url}/pdfs?date=#{@date}",
+                             headers: {
+                               Authorization: auth_token
+                             }).body
+        Zip::InputStream.open(StringIO.new(input)) do |io|
+          while entry = io.get_next_entry #note: there is only actually one pdf here
+            pdf = io.read
+            filename = entry.name
+            Bucket.new.put_object("#{s3_path}/#{filename}", pdf)
+          end
+        end
+      end
+
+      def save
+        bookings_hash = json
+        ::OkcBlotter::Pdf.create!(
+          parsed_on: DateTime.now,
+          date: @date,
+          bookings: bookings_hash.map do |booking_hash|
+            build_bookings(booking_hash)
+          end
```

Comment:
here we're using `.map` to loop through the bookings hash and call the `build_bookings` method to build an object for each item in the hash 
(map is like each but it creates a new array with the returned results)

Reply:
Got it!